src/coreclr/nativeaot/Runtime/StackFrameIterator.cpp

```diff
@@ -1830,7 +1825,10 @@ void StackFrameIterator::PrepareToYieldFrame()
     ASSERT(m_pInstance->IsManaged(m_ControlPC) ||
          ((m_dwFlags & SkipNativeFrames) == 0 && (m_dwFlags & UnwoundReversePInvoke) != 0));
 
-    if (m_dwFlags & ApplyReturnAddressAdjustment)
+    // Do not adjust the PC if ExCollide is set since in that case it's set to the
+    // beginning of the instruction from ExInfo and adjusting it would cause to move
+    // to the previous instruction.
```

Comment:
```suggestion
    // Do not adjust the PC if ExCollide is set. In that case it was either copied from
    // existing stack frame iterator that was already adjusted, or it's set to the
    // beginning of the faulting instruction from ExInfo and adjusting it would cause
    // us to move to the previous instruction which may no longer be in the same try
    // block.
```